include/camp/resource/cuda.hpp

```diff
@@ -129,20 +187,19 @@ namespace resources
         // related: https://stackoverflow.com/questions/64523302/cuda-missing-return-statement-at-end-of-non-void-function-in-constexpr-if-fun
         return MemoryAccess::Unknown;
       }
-    public:
-      Cuda(int group = -1, int dev = 0)
-          : stream(get_a_stream(group)), device(dev)
-      {
-      }
 
-      /// Create a resource from a custom stream
-      /// The device specified must match the stream, if none is specified the
-      /// currently selected device is used.
+    public:
+      explicit Cuda(int group = -1, int dev = get_current_device())
+          : stream(get_a_stream(group, dev)), device(dev)
+      { }
+
+      /// Create a resource from a custom stream.
+      /// If device is specified it must match the stream. If device is
+      /// unspecified, we will get it from the stream.
+      /// This may be called before main if device is specified as no calls to
```

Comment:
We had an issue recently that took a while to resolve where we were making Cuda runtime library calls before entering main(). Will this present similar problems, potentially?

Reply:
This coding was the source of that problem. We ended up changing the way that we create some resources to avoid the issue that works before and after these changes.

Reply:
@MrBurmark, I just happened back across this.  Would it have solved the issue to have the streams bound lazily?